tests/src/test/scala/org/http4s/multipart/MultipartSpec.scala

```diff
@@ -1,77 +1,85 @@
         encoded and decoded without content types  $encodeAndDecodeMultipartMissingContentType
-        encoded and decoded with    binary data    $encodeAndDecodeMultipartWithBinaryFormData        
+        encoded and decoded with    binary data    $encodeAndDecodeMultipartWithBinaryFormData
         decode  and encode  with    content types  $decodeMultipartRequestWithContentTypes
         decode  and encode  without content types  $decodeMultipartRequestWithoutContentTypes
      """
+
   val url = Uri(
       scheme = Some(CaseInsensitiveString("https")),
       authority = Some(Authority(host = RegName("example.com"))),
       path = "/path/to/some/where")
 
-  implicit lazy val MultiPartEq: Equal[Multipart] =
-    Equal.equalBy[Multipart, Vector[Part]](_.parts)
+  def toBV(entityBody: EntityBody): ByteVector = ByteVector(entityBody.runLog.unsafeRun())
 
-  // a.headers == b.headers doesn't do what I expected.
-  implicit lazy val PartEq: Equal[Part] =
-    Equal.equal { (a, b) => a.headers.size == b.headers.size && (a.headers zip b.headers).forall { case (ah, bh) => ah == bh } && a.body === b.body }
+  implicit def partEq : Eq[Part] = Eq.instance[Part] { case (a, b) =>
+    a.headers === b.headers &&
+      {
+        for {
+          abv <- a.body.runLog.map(ByteVector(_))
+          bbv <- b.body.runLog.map(ByteVector(_))
+        } yield abv === bbv
+      }.unsafeRun()
+
+  }
+
+  implicit val multipartEq : Eq[Multipart] = Eq.instance[Multipart]{ (a, b) =>
+    a.headers === b.headers &&
+      a.boundary === b.boundary &&
+      a.parts === b.parts
+  }
 
-  // This one is shady.
-  implicit lazy val EntityBodyEq: Equal[EntityBody] =
-    Equal.equalBy[EntityBody, String](_.pipe(utf8Decode).runFoldMap(identity).run)
 
   def encodeAndDecodeMultipart = {
 
     val field1     = Part.formData("field1", "Text_Field_1", `Content-Type`(`text/plain`))
     val field2     = Part.formData("field2", "Text_Field_2")
     val multipart  = Multipart(Vector(field1,field2))
     val entity     = EntityEncoder[Multipart].toEntity(multipart)
-    val body       = entity.run.body
+    val body       = entity.unsafeRun().body
     val request    = Request(method  = Method.POST,
                              uri     = url,
                              body    = body,
                              headers = multipart.headers )
     val decoded    = EntityDecoder[Multipart].decode(request, true)
-    val result     = decoded.run.run
-    
-    result must beRight.like { case mp => mp must beTypedEqualTo(multipart, Equal[Multipart].equal) }
+    val result     = decoded.value.unsafeRun
+
+    result must beRight.like { case mp =>
+      mp === multipart
+//            mp.headers must_=== multipart.headers
+//            mp.parts.map(_.headers) must_=== multipart.parts.map(_.headers)
+//      mp.parts.map(_.body.map(_.decodeUtf8).unsafeRun) must_=== multipart.parts.map(_.body.map(_.decodeUtf8).unsafeRun)
```

Comment:
Not really, it survived as it let me test parts separate, now that Eq instances are in scope this could be axed.